src/table/src/table.rs

```diff
@@ -113,3 +143,84 @@ impl Table {
             .map(|(_, c)| c.clone())
     }
 }
+
+/// Collects column default [`Expr`] from column schemas.
+fn collect_column_defaults(column_schemas: &[ColumnSchema]) -> HashMap<String, Expr> {
+    column_schemas
+        .iter()
+        .filter_map(|column_schema| {
+            default_constraint_to_expr(
+                column_schema.default_constraint()?,
+                &column_schema.data_type,
+            )
+            .map(|expr| (column_schema.name.to_string(), expr))
+        })
+        .collect()
+}
+
+/// Try to cast the [`ColumnDefaultConstraint`] to [`Expr`] by the target data type.
+fn default_constraint_to_expr(
+    default_constraint: &ColumnDefaultConstraint,
+    target_type: &ConcreteDataType,
+) -> Option<Expr> {
+    match default_constraint {
+        ColumnDefaultConstraint::Value(v) => {
+            v.try_to_scalar_value(target_type).ok().map(Expr::Literal)
+        }
+
+        ColumnDefaultConstraint::Function(name)
+            if matches!(
+                name.as_str(),
+                CURRENT_TIMESTAMP | CURRENT_TIMESTAMP_FN | NOW_FN
+            ) =>
+        {
+            Some(Expr::Cast(Cast {
+                expr: Box::new(NOW_EXPR.clone()),
```

Comment:
Looks like we only need a convenient builder for `now()` UDF but not a lazy static?

Reply:
We just want to reuse it, and try to avoid creating it by `SessionContext` repeatedly.